src/Features/Core/Portable/CodeStyle/AbstractCodeStyleProvider.Refactoring.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    // This part contains all the logic for hooking up the CodeRefactoring to the CodeStyleProvider.
+    // All the code in this part is an implementation detail and is intentionally private so that
+    // subclasses cannot change anything.  All code relevant to subclasses relating to refactorings
+    // is contained in AbstractCodeStyleProvider.cs
+
+    internal abstract partial class AbstractCodeStyleProvider<TOptionKind, TCodeStyleProvider>
+    {
+        private async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var optionValue = optionSet.GetOption(_option);
+
+            var severity = GetOptionSeverity(optionValue);
+            switch (severity)
+            {
+                case ReportDiagnostic.Suppress:
+                case ReportDiagnostic.Hidden:
```

Comment:
I'm definitely a bit confused here.  My understanding of hte hierarchy was:

None: Completely disabled.  
Hidden/Silent: Enabled, but only exposed as a refactoring (i.e. a screwdriver, not as a lightbulb)
Info/Warning/Error: Enabled, exposed as a lightbulb.

Reply:
Hidden severity diagnostics are *supposed* to use the screwdriver. They are refactorings that perform analysis in advance and can offer Fix All, and perhaps most importantly, can be implemented in NuGet packages.

Reply:
>Hidden severity diagnostics are supposed to use the screwdriver. They are refactorings that perform analysis in advance and can offer Fix All, and perhaps most importantly, can be implemented in NuGet packages.

Note: *currently* i think there is no way to accomplish that.  Specifically 'refactorings' (i.e. things provided by CodeRefactoringProvider) cannot have FixAll.  After all, they have no FixAllProvider exposed on them to actually perform the fix-all.

Conversely, all CodeFixProviders (which can have fix-alls) are always shown as lightbulbs not screwdrivers.

--

Would yo ube willing to take this offline to gitter?  This area is a bit confusing, and it might be easier to hash things out 1:1 to get totaly clarity on how this all works now, what we can do right now, and where we want to get to in the future.